rest/admin_api.go

```diff
@@ -915,19 +915,37 @@ func (h *handler) handlePutDbConfigImportFilter() error {
 // In non-persistent mode, the endpoint just removes the database from the node.
 func (h *handler) handleDeleteDB() error {
 	h.assertAdminOnly()
+	dbName := h.PathVar("olddb")
+
+	var bucket string
+	//var buckets []string
+
+	dbContext, _ := h.server.GetDatabase(h.ctx(), dbName)
+
+	if dbContext != nil {
+		bucket = dbContext.Bucket.GetName()
+	} else if h.server.persistentConfig {
+		_, cnf, err := h.server.fetchDatabase(h.ctx(), dbName)
+		if err != nil {
+			return err
+		}
+		bucket = *cnf.Bucket
```

Comment:
Maybe do a defensive nil check here on cnf and cnf.Bucket to avoid panicing if we haven't got either.